src/api/spec/models/user/has_local_permission_spec.rb

```diff
@@ -0,0 +1,101 @@
+RSpec.describe User, '.has_local_permission?' do
+  subject { user.has_local_permission?(perm_string, object) }
+
+  let(:user) { create(:confirmed_user) }
+
+  describe 'package' do
+    let(:object) { create(:package) }
+    let(:perm_string) { 'change_package' }
+
+    it { expect(subject).to be_falsey }
+
+    context 'has user relationship' do
+      let(:object) { create(:package_with_maintainer, maintainer: user) }
+
+      it { expect(subject).to be_truthy }
+    end
+
+    context 'has group relationship' do
+      let(:group) { create(:group_with_user, user: user) }
+      let(:object) { create(:package_with_maintainer, maintainer: group) }
+
+      it { expect(subject).to be_truthy }
+    end
+
+    context 'is in user home project' do
+      let(:project) { create(:project, name: "#{user.home_project_name}:hans") }
+      let(:object) { create(:package, project: project) }
+
+      it { expect(subject).to be_truthy }
+    end
+
+    context 'is in user home sub-project' do
+      let(:project) { create(:project, name: user.home_project_name) }
+      let(:object) { create(:package, project: project) }
+
+      it { expect(subject).to be_truthy }
+    end
```

Comment:
```suggestion
    context 'is in user home project' do
      let(:project) { create(:project, name: user.home_project_name) }
      let(:object) { create(:package, project: project) }

      it { expect(subject).to be_truthy }
    end

    context 'is in user home sub-project' do
      let(:project) { create(:project, name: "#{user.home_project_name}:hans") }
      let(:object) { create(:package, project: project) }

      it { expect(subject).to be_truthy }
    end
```
This is correct for the `describe 'project' do` section below.

Reply:
Fixed in #16517.